### PR TITLE
8253585: Add basic vector mask implementation for Arm SVE

### PR DESCRIPTION
**Pull request by XiaohongGong**
Summary: Add the SVE match rule support for vector mask ops.

======

Currently the vector mask related operations like vector mask load/store, vector mask compare and vector blend are not supported for Arm SVE. This patch adds the basic backend implementation for them.

BTW, this is a basic version for vector mask operations that does not use the SVE predicate features. And we will make it happen in future.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Testing

|     | Linux x32 | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (5/5 passed) | ❌ (2/2 failed) | ✔️ (2/2 passed) |
| Test (tier1) |    |  ✔️ (9/9 passed) |    |  ✔️ (9/9 passed) |

**Failed test tasks**
- [Windows x64 (build debug)](https://github.com/XiaohongGong/panama-vector/runs/1356248508)
- [Windows x64 (build release)](https://github.com/XiaohongGong/panama-vector/runs/1356248501)

### Issue
 * [JDK-8253585](https://bugs.openjdk.java.net/browse/JDK-8253585): Add basic vector mask implementation for Arm SVE

### Reviewers
 * [Ningsheng Jian](https://openjdk.java.net/census#njian) (@nsjian - Committer)
 * [Wang Huang](https://openjdk.java.net/census#whuang) (@Wanghuang-Huawei - no project role)

### Download
`$ git fetch https://git.openjdk.java.net/panama-vector pull/20/head:pull/20`
`$ git checkout pull/20`

### CODE REVIEW

**Comment by XiaohongGong** [comment]
Hi, could anyone please take a look at this PR? Thanks!

**Review by nsjian — COMMENTED** [review]
LGTM (I have reviewed it internally).

File: src/hotspot/cpu/aarch64/aarch64-asmtest.py [hunk 1]
```diff
@@ -1479,55 +1479,101 @@ def generate(kind, names):
...
-                        ["stlxpw", "__ stlxpw(r2, zr, zr, r3);",                         "stlxp\tw2, wzr, wzr, [x3]"],
-                        ["stxp",   "__ stxp(r4, zr, zr, r5);",                           "stxp\tw4, xzr, xzr, [x5]"],
-                        ["stxpw",  "__ stxpw(r6, zr, zr, sp);",                          "stxp\tw6, wzr, wzr, [sp]"],
-                        ["dup",    "__ dup(v0, __ T16B, zr);",                           "dup\tv0.16b, wzr"],
-                        ["mov",    "__ mov(v1, __ T1D, 0, zr);",                         "mov\tv1.d[0], xzr"],
-                        ["mov",    "__ mov(v1, __ T2S, 1, zr);",                         "mov\tv1.s[1], wzr"],
-                        ["mov",    "__ mov(v1, __ T4H, 2, zr);",                         "mov\tv1.h[2], wzr"],
-                        ["mov",    "__ mov(v1, __ T8B, 3, zr);",                         "mov\tv1.b[3], wzr"],
-                        ["ld1",    "__ ld1(v31, v0, __ T2D, Address(__ post(r1, r0)));", "ld1\t{v31.2d, v0.2d}, [x1], x0"],
+generate(SpecialCases, [["ccmn",    "__ ccmn(zr, zr, 3u, Assembler::LE);",                "ccmn\txzr, xzr, #3, LE"],
```
[the remark this answers is not in the document]
Comment (XiaohongGong) [reply src/hotspot/cpu/aarch64/aarch64-asmtest.py:1482]: Hi @Wanghuang-Huawei thanks for looking at this PR. The space is added to keep the row alignment with the new added cases from "uunpkhi" to "whilels".

**Review by Wanghuang-Huawei — APPROVED** [review]

**Comment by XiaohongGong** [comment]
> LGTM

Thanks for the review!

**Review by nsjian — APPROVED** [review]

**Pull request closed without merging**